### ISSUE TITLE
Please add more documentation to Model reload 

### ISSUE THREAD

**Issue by mhankus** [emberjs/data#9613]
### Description
 Currently docs for Model.reload does not specify how reload works when object is dirty (hasDirtyAttributes returns true).  I was sure that reload was discarding all changes made to record, and load fresh state from server. Right now I see that changes need to be rolledback.  Please add clarification to documentation what happens in that case.
 
### Versions

Run the following command and paste the output below: `pnpm list ember-source && pnpm list ember-cli && pnpm list "*ember-data*"`.

```cli
└─ ember-cli@5.12.0
└─ ember-source@5.12.0
└─ ember-data@5.3.9
```

**Comment by runspired** [comment]
correct, reload will update remote state only. It will not discard any mutations you've made. I'm not sure its worth adjusting documentation as this API is scheduled for deprecation and `model.reload` works the same as any other request in this way.
